### PR TITLE
Fixed search view colors

### LINKED ISSUE #3849: Fix visual issues in Magenta theme
The new "experimental" Magenta theme has some visual issues that will need to be fixed before it's ready for primetime:

- [x] Some dialogs in the app use the wrong button styles:

- [x] Search bar has extra 🔍 icon in "Delete saved form" search

- [x] Search view text colors are wrong (see screenshot above)

- [x] On magenta view scroll is not visible until you actually start to scroll what differentiates it from the other modes

It's totally fine if picking this up to just solve one of the issues!

### PR DESCRIPTION
**Pull request by grzesiek2010**
Work towards #3849 

#### What has been done to verify that this works as intended?
I checked the search view in all three themes:

|  |  |
| ------------- | ------------- |
| Screenshot_1592571326 | Screenshot_1592571338  |

#### Why is this the best possible solution? Were any other approaches considered?
I tried adapting colors in themes but it didn't work so I had to do that problematically.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It's not a risky change but requires testing the search view in all three themes.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [ ] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by seadowg** [comment]
Just changed the description to "Work towards" from "Closes" as we might have a few more dialogs to fix.

File: collect_app/src/main/res/values/magenta_theme.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
         <item name="colorOnBackground">#000000</item>
         <item name="colorOnError">#FFFFFF</item>
         <item name="colorOnSurface">#000000</item>
+        <item name="android:textColorHint">#E3E3E3</item>
```
Comment (seadowg) [inline collect_app/src/main/res/values/magenta_theme.xml]: I'm guessing this corrects the hint color for search? If so, I think we should have a comment here to make sure that's clear for when we're making a new theme or changing the color.

**Review by seadowg — CHANGES_REQUESTED** [review]

**Comment by seadowg** [comment]
Just one comment about a comment but looking good (we're so close to this being able to go out 😃). I think this can go to QA as we shouldn't need any actual code changes.

@getodk-bot label "needs testing"

File: collect_app/src/main/res/values/magenta_theme.xml (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/res/values/magenta_theme.xml]: Done.

**Review by seadowg — APPROVED** [review]

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Androids: 5.1, 7.1, 8.1, 9.0 and 10. 

Tested scenarios:
- search view on: Fill Blank Form, Edit Saved Form, Send Finalized Form, View Sent Form, Get Blank Form and both Delete Saved Form.
- light mode
- dark mode
- magenta mode

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**